streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -216,11 +217,13 @@ public interface UserTaskAssignmentListener {
     private Queue<StreamsException> nonFatalExceptionsToHandle;
     private Time time;
 
+    // since live upgrades from 2.3- to 4.0 or above are no longer supported, we can always
```

Comment:
```suggestion
    // since live upgrades from 2.3 (or earlier) to 4.0 or above are no longer supported, we can always
```
The minus sign easily reads as a typo.